File: ThucHanh1/Bai6/app/src/main/java/TH/Bai6/MainActivity.java
package TH.Bai6;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.google.android.material.textfield.TextInputEditText;

public class MainActivity extends AppCompatActivity {

    Button submitBtn;
    TextInputEditText heightInput;
    TextInputEditText weightInput;
    TextInputEditText resultInput;
    TextInputEditText danhGia;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        initViews();
        listener();
    }

    private void initViews () {
        submitBtn = findViewById(R.id.submitbtn);
        heightInput = findViewById(R.id.height);
        weightInput = findViewById(R.id.weight);
        resultInput = findViewById(R.id.result);
        danhGia = findViewById(R.id.danhgia);
    }

    private void listener() {
        submitBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String heightValue = heightInput.getText().toString().trim();
                String weightValue = weightInput.getText().toString().trim();

                float result = (float) (
                        Float.parseFloat(weightValue) /
                                Math.pow(Float.parseFloat(heightValue), 2)
                );

                resultInput.setText(String.valueOf(result));

                if (result < 18.5) {
                    danhGia.setText("Gầy");
                } else if (result >= 18.5 && result < 25) {
                    danhGia.setText("Bình thường");
                } else {
                    danhGia.setText("Béoooooooooooooooooooooo");
                }
            }
        });
    }
}

File: ThucHanh2/Bai2/app/src/main/java/com/example/bai2/MainActivity.kt
package com.example.bai2
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }
    fun onclickTong(v: View?) {
        //Khai báo các biến điều khiển tương ứng
        val height = findViewById<EditText>(R.id.height)
            .text.toString().toFloatOrNull()
        val weight = findViewById<EditText>(R.id.weight).text.toString().toFloatOrNull()


        if(height!= null && weight !=null){
            var BMI = weight/(height*height)
            val result: TextView = findViewById(R.id.Result)
            when (BMI) {
                in 0.0..18.49 -> result.text = "Chỉ số BMI của bạn là $BMI đang ở mức Bạn quá gầy"
                in 18.5..24.9 -> result.text = "Chỉ số BMI của bạn là $BMI đang ở mức Bình thường"
                in 25.0..29.9 -> result.text = "Chỉ số BMI của bạn là $BMI đang ở mức Tiền béo phì"
                in 30.0..34.9 -> result.text = "Chỉ số BMI của bạn là $BMI đang ở mức Béo phì độ I"
                in 35.0..39.9 -> result.text = "Chỉ số BMI của bạn là $BMI đang ở mức Béo phì độ II"
                else -> result.text = "Chỉ số BMI của bạn là $BMI đang ở mức Béo phì độ III"
            }
        }
    }
}